### PR TITLE
Switch to Jakarta

### PR DESCRIPTION
**Pull request by tomas-langer**
Related to #137 

Changes for Jakarta packages and MP Parent for `master` branch.

I think I cannot create the branch in this repo to do an RC release, if this PR is approved, could somebody from the maintainers kindly do that? Thanks!

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java [hunk 1]
```diff
@@ -176,8 +176,7 @@ public T getPayload() {
         Function<Throwable, CompletionStage<Void>> nack = getNack();
         if (nack == null) {
             return CompletableFuture.completedFuture(null);
-        }
-        else {
+        } else {
```
Comment (cescoffier) [inline api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java:179]: Cosmetic - Any reason to change the format?

**Review by cescoffier — APPROVED** [review]
Except the formatting change (and the removal of the config), it looks good to me. I'm going to run some tests on our side to see where we stand.

**Comment by cescoffier** [comment]
Building the branch seems to update the format of almost every single file.

**Comment by radcortez** [comment]
> Building the branch seems to update the format of almost every single file.

Yes, the parent pom now applies a consistent code style to every MP project. Before it was per project.

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/package-info.java [hunk 2]
```diff
@@ -53,5 +53,5 @@
  * </pre>
  */
 
-@org.osgi.annotation.versioning.Version("1.1")
+@org.osgi.annotation.versioning.Version("3.0")
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/reactive/messaging/package-info.java:56]: The new version should be 2.0 not 3.0. Every package version should tell whether there are changes in a release. This version is not the version of spec.

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/package-info.java [hunk 3]
```diff
@@ -39,5 +39,5 @@
  * {@link org.eclipse.microprofile.reactive.messaging.spi.Connector} qualifier. Connectors receive the channel
  * configuration matching their {@link Connector} name.
  */
-@org.osgi.annotation.versioning.Version("1.1")
+@org.osgi.annotation.versioning.Version("3.0")
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/package-info.java:42]:
```suggestion
@org.osgi.annotation.versioning.Version("2.0")
```

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/package-info.java (hunk 2 above)
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/reactive/messaging/package-info.java:56]:
```suggestion
@org.osgi.annotation.versioning.Version("2.0")
```

File: pom.xml [hunk 4]
```diff
@@ -22,33 +22,33 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
+    <parent>
+        <groupId>org.eclipse.microprofile</groupId>
+        <artifactId>microprofile-parent</artifactId>
+        <version>2.1</version>
```
Comment (Emily-Jiang) [inline pom.xml:28]:
```suggestion
        <version>2.3</version>
```

File: pom.xml [hunk 5]
```diff
@@ -22,33 +22,33 @@
...
         <!-- whether autorelease maven central staging repositories - default true to allow review and manually release repositories -->
         <autorelease>false</autorelease>
         <!-- keeping closed repos with failure - default is false because the errors are visible in the maven output, but true will leave the repo open for investigation in Sonatype Nexus -->
         <keepStagingReposOnFailure>false</keepStagingReposOnFailure>
-        <microprofile-config-api.version>2.0</microprofile-config-api.version>
-        <microprofile-metrics-api.version>3.0</microprofile-metrics-api.version>
-        <cdi-api.version>2.0</cdi-api.version>
+        <microprofile-config-api.version>3.0</microprofile-config-api.version>
+        <microprofile-metrics-api.version>4.0</microprofile-metrics-api.version>
+        <cdi-api.version>3.0</cdi-api.version>
```
Comment (Emily-Jiang) [inline pom.xml:51]:
```suggestion
```

File: pom.xml [hunk 6]
```diff
@@ -149,63 +149,16 @@
...
-                <artifactId>javax.json.bind-api</artifactId>
-                <version>1.0</version>
-            </dependency>
-            <dependency>
-                <groupId>org.jboss.arquillian</groupId>
-                <artifactId>arquillian-bom</artifactId>
-                <version>1.6.0.Final</version>
-                <scope>import</scope>
-                <type>pom</type>
-            </dependency>
```
Comment (Emily-Jiang) [inline pom.xml:208]: The section `<build>` should not be necessary. Please just get rid of them as the parent pom.xml has them.

File: tck/pom.xml [hunk 7]
```diff
@@ -25,7 +25,7 @@
     <parent>
         <groupId>org.eclipse.microprofile.reactive.messaging</groupId>
         <artifactId>microprofile-reactive-messaging-parent</artifactId>
-        <version>2.1-SNAPSHOT</version>
+        <version>3.0-SNAPSHOT</version>
```
Comment (Emily-Jiang) [inline tck/pom.xml:28]: The arquillian dependency management is incorrect. The correct version has been specified in the parent pom.xml. You just need to declare the dependency on the api and remove the line 35 and 39-40

**Review by Emily-Jiang — CHANGES_REQUESTED** [review]
See the inline comments. Also the spec needs to have a release note to document the changes in 2.0.

**Comment by Emily-Jiang** [comment]
A marker file Tck needs to be added to the tck folder. See an example [here](https://github.com/eclipse/microprofile-fault-tolerance/tree/master/tck/src/main/resources/META-INF)

**Comment by Emily-Jiang** [comment]
Step 4, 5 and 6 needs be done in [this](https://docs.google.com/document/d/1bhhfmhxHJpQdmdUtQw5o_WtScpzkfHoTNGPvrQnGhc0/edit#) doc.

**Review by Emily-Jiang — APPROVED** [review]
LGTM!

**Pull request merged**